### PR TITLE
Get master-next building again.

### PR DESCRIPTION
**Pull request by Teemperor**
Series of commits to keep up with LLVM/Clang API changes.

### CODE REVIEW

**Comment by Teemperor** [comment]
@swift-ci test

File: lib/IRGen/GenStruct.cpp [hunk 1]
```diff
@@ -26,6 +26,7 @@
 #include "llvm/IR/DerivedTypes.h"
 #include "llvm/IR/Function.h"
 #include "clang/AST/ASTContext.h"
+#include "clang/AST/Attr.h"
```
Comment (dcci) [inline lib/IRGen/GenStruct.cpp]: Why do you need this?

Comment (dcci) [reply lib/IRGen/GenStruct.cpp]: OK, I see now.

**Review by dcci — APPROVED** [review]

**Comment by Teemperor** [comment]
lldb fails to build but at least we can build swift now. Merging this that I can make the lldb PR.

**Pull request merged**